### PR TITLE
feat: add spam protection via honeypot

### PR DESCRIPTION
**Pull request by AmoabaKelvin**
when a form is filled by a bot, it contains the key `_gotcha` in the formdata, using that, we can identify that the form was submitted by a bot and then flag it as a spam submission.

### CODE REVIEW

**Comment by aikins01** [comment]
😂😂😂 gotcha, going through

**Review by aikins01 — COMMENTED** [review]
When does the button appear

**Comment by AmoabaKelvin** [comment]
> When does the button appear

Button, what do you mean 🤔

File: src/app/(main)/form/[id]/submissions-table.tsx [hunk 1]
```diff
@@ -260,6 +267,10 @@ export function SubmissionsTable({
                 <TableRow
                   key={row.id}
                   data-state={row.getIsSelected() && "selected"}
+                  className={cn(
+                    row.original.isSpam &&
+                      "bg-red-100 duration-300 hover:bg-red-50",
```
Comment (aikins01) [inline src/app/(main)/form/[id]/submissions-table.tsx:272]: oh, it's the on hover classes that got me here. I was thinking a new button was added

**Review by aikins01 — APPROVED** [review]
Looks good 👍

File: src/app/(main)/form/[id]/submissions-table.tsx [hunk 2]
```diff
@@ -260,6 +267,10 @@ export function SubmissionsTable({
                 <TableRow
                   key={row.id}
                   data-state={row.getIsSelected() && "selected"}
+                  className={cn(
```
Comment (dephraiim) [inline src/app/(main)/form/[id]/submissions-table.tsx:270]: this isn't how to use `cn`

```js
cn('base class stuff', {
	'class stuff': isLoggedIn,
})

// proper example or something related
className={cn({
  'bg-red-100 duration-300 hover:bg-red-50': row.original.isSpam
})}
```

**Review by dephraiim — CHANGES_REQUESTED** [review]

File: src/app/(main)/form/[id]/submissions-table.tsx (hunk 2 above)
Comment (aikins01) [reply src/app/(main)/form/[id]/submissions-table.tsx:270]: TIL 😅😅

**Pull request closed without merging**